frontend/src/metabase/lib/pulse.js

```diff
@@ -8,6 +8,28 @@ export function channelIsValid(channel, channelSpec) {
   if (!channelSpec) {
     return false;
   }
+
+  if (channelSpec.recipients) {
+    // default from formInput is an empty array, not a null array
+    // check for both
+    if (!channel.recipients || channel.recipients.length < 1) {
+      return false;
+    }
+  }
+
+  if (channelSpec.fields) {
+    for (const field of channelSpec.fields) {
+      if (
+        field.required &&
+        channel.details &&
+        (channel.details[field.name] == null ||
+          channel.details[field.name] === "")
+      ) {
+        return false;
+      }
+    }
+  }
+
```

Comment:
FYI, didn't go into rewriting the logic or simplifying `channelIsValid` function in any way for this PR.

Before this PR the `switch` statement was executed before the logic that checks whether there are any recipients or not. For whatever reason, that `switch` statement returns `true` if the selected schedule was `monthly` (which is exactly the error @flamber discovered and described in the issue).

Reply:
Makes sense